src/GTP.cpp

```diff
@@ -161,6 +161,7 @@ std::string GTP::get_life_list(const GameState & game, bool live) {
 
 bool GTP::execute(GameState & game, std::string xinput) {
     std::string input;
+    static UCTSearch search;
```

Comment:
You could make this a unique_ptr<UCTSearch>. The reason for doing so would be that it avoids putting a potentially very big object on the stack, and it reduces the changes you need to make in GTP::execute as well.

(This has become relevant since it now carries a full copy of GameState, rather than just a reference)

Reply:
done.